### PR TITLE
Issue #397: Replace uc_store_page_alter() with a footer block.

### LINKED ISSUE #397: Restore Ubercart footer functionality previously provided by uc_store_page_alter().
EDIT: original title was "uc_store_page_alter() doesn't do anything".

In `uc_store.module`, the function `uc_store_page_alter()` implements `hook_page_alter()`, but that is not a Backdrop hook.

Also `uc_cybersource.module` implements this hook in `uc_cybersource_page_alter()`.

### PR DESCRIPTION
**Pull request by bugfolder**
Fixes https://github.com/backdrop-contrib/ubercart/issues/397.

### CODE REVIEW

File: uc_store/classes/uc_layout_access.inc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  function checkAccess() {
+    $path = backdrop_get_normal_path($_GET['q']);
+    $parts = explode('/', $path);
+
+    // Check whether the page is governed by Ubercart.
+    $page_match == FALSE;
```
Comment (argiepiano) [inline uc_store/classes/uc_layout_access.inc]: This should be `$page_match = FALSE;`

File: uc_store/uc_store.module [hunk 2]
```diff
@@ -644,64 +656,38 @@ function uc_store_date_format_types() {
...
-  }
-
   $messages = _uc_store_footer_options();
-
+  $id = config_get('uc_store.settings', 'uc_footer_message');
   if ($id == 0) {
     // Pseudorandom number based on the hash of the path and the site's private
     // key, so messages are consistent between pages on the same site, but
     // different on the same pages on different sites.
     $id = (hexdec(substr(md5($path . backdrop_get_private_key()), 0, 2)) % count($messages)) + 1;
```
Comment (argiepiano) [inline uc_store/uc_store.module]: $path is now undefined.

File: uc_store/classes/uc_layout_access.inc (hunk 1 above)
Comment (bugfolder) [reply uc_store/classes/uc_layout_access.inc]: Thanks, fixed.

File: uc_store/uc_store.module (hunk 2 above)
Comment (bugfolder) [reply uc_store/uc_store.module]: Ah, overlooked that. Fixed, thanks.

File: uc_store/classes/uc_layout_access.inc [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      'paths' => '',
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  function summary() {
+    if ($this->settings['visibility_setting']) {
+      return t('Current page is an Ubercart page.');
```
Comment (argiepiano) [inline uc_store/classes/uc_layout_access.inc]: Is it possible to also include any additional paths in the summary?

Comment (bugfolder) [reply uc_store/classes/uc_layout_access.inc]: Yup, added.

**Pull request merged**